govcd/edgegateway.go

```diff
@@ -287,22 +287,47 @@ func (eGW *EdgeGateway) AddDNATRule(ruleDetails NatRule) ([]*types.NatRule, erro
 		return nil, fmt.Errorf("%s", combinedTaskErrorMessage(task.Task, err))
 	}
 
+	var createdNatRule *types.NatRule
+
 	err = eGW.Refresh()
 	if err != nil {
 		return nil, fmt.Errorf("error refreshing edge gateway: %#v", err)
 	}
 
-	return eGW.EdgeGateway.Configuration.EdgeGatewayServiceConfiguration.NatService.NatRule, nil
+	for _, natRule := range eGW.EdgeGateway.Configuration.EdgeGatewayServiceConfiguration.NatService.NatRule {
+		if natRule.Description == mappingId {
+			createdNatRule = natRule
+			break
+		}
+	}
+
+	if createdNatRule == nil {
+		return nil, fmt.Errorf("error creating DNAT rule, didn't match created rule")
+	}
+
+	createdNatRule.Description = originalDescription
+
+	return eGW.UpdateNatRule(createdNatRule)
 }
 
-// AddSNATRule creates SNAT rule and returns refreshed existing NAT rules array or error
-// Allows to assign specific network Org VDC or external. Old function AddNATPortMapping and
-// AddNATMapping assigns rule to first external network
-func (eGW *EdgeGateway) AddSNATRule(networkHref, externalIP, internalIP, description string) ([]*types.NatRule, error) {
+// AddSNATRule creates SNAT rule and returns created NAT rule or error.
+// Allows to assign specific Org VDC or external network.
+// Old function AddNATPortMapping and AddNATMapping assigns rule to only first external network
+func (eGW *EdgeGateway) AddSNATRule(networkHref, externalIP, internalIP, description string) (*types.NatRule, error) {
+
+	// As vCD API don't return rule API we fetch it manually:
+	//  * create rule with description which value is our generated Id
+	//  * find rule which has description with our generated Id
```

Comment:
Please add additional bullet:
`* read the real (vCD's) rule Id`

Reply:
done